src/components/interaction-designer/block-editors/choices/SelectOneResponseBlockContactPropertyEditor.vue

```diff
@@ -0,0 +1,182 @@
+<template>
+  <div class="select-one-response-block-contact-property-editor">
+    <generic-contact-property-editor
+      :block="block"
+      @toggleSetContactProperty="onSetContactPropertyToggle"
+      @changeContactPropertyType="resetMapping"
+      @updateShouldUseCurrentBlockResponse="onShouldUseCurrentBlockResponseUpdate">
```

Comment:
I'm not a big fan of the chosen approach - of reusing GenericContactPropertyEditor. Maybe it would've been better to break it down to smaller parts and reuse _them_ in this file. I don't know.

Reply:
That's why I initially thoughts about having a similar component without reusing it. Feel free to refactor if you find that cleaner